File: 2022/StudioMallup/app/src/main/java/com/creator/studiomallup/util/HNConfig.java
package com.creator.studiomallup.util;

import android.content.Context;
import android.content.pm.PackageManager;

import com.creator.studiomallup.common.HNApplication;
import com.creator.studiomallup.delegator.HNSharedPreference;


/**
 * Created by skcrackers on 8/5/16.
 */
public class HNConfig {
    private Context ctx;

    public HNConfig(Context paramContext)
    {
        this.ctx = paramContext;
    }

    public String getAppVersion() {
        try {
            String str = ctx.getPackageManager().getPackageInfo(ctx.getPackageName(), 0).versionName;
            return str;
        } catch (PackageManager.NameNotFoundException localNameNotFoundException) {
        }
        return "";
    }

    public String getDeviceId() {
        LogUtil.d("getDeviceId() : " + HNApplication.mDeviceId);
        return HNApplication.mDeviceId;
    }

    public String getMemberKey() {
        return HNSharedPreference.getSharedPreference(ctx, "memberKey");
    }
    public void setMemberKey(String paramString) {
        HNSharedPreference.putSharedPreference(ctx, "memberKey", paramString);
    }

    public String getAppId() {
        return HNSharedPreference.getSharedPreference(ctx, "appId");
    }
    public void setAppId(String paramString) {
        HNSharedPreference.putSharedPreference(ctx, "appId", paramString);
    }

    public String getUserId() {
        return HNSharedPreference.getSharedPreference(ctx, "userId");
    }
    public void setUserId(String paramString) {
        HNSharedPreference.putSharedPreference(ctx, "userId", paramString);
    }

    public String getChannelId() {
        return HNSharedPreference.getSharedPreference(ctx, "channelId");
    }
    public void setChannelId(String paramString) {
        HNSharedPreference.putSharedPreference(ctx, "channelId", paramString);
    }

    public String getRequestId() {
        return HNSharedPreference.getSharedPreference(ctx, "requestId");
    }
    public void setRequestId(String paramString) {
        HNSharedPreference.putSharedPreference(ctx, "requestId", paramString);
    }
}

File: 2023/MallupLive/app/src/main/java/com/creator/malluplive/util/HNConfig.kt
package com.creator.malluplive.util

import android.content.Context
import android.content.pm.PackageManager
import com.creator.malluplive.common.HNApplication
import com.creator.malluplive.delegator.HNSharedPreference

/**
 * Created by skcrackers on 8/5/16.
 */
class HNConfig(private val ctx: Context) {
    val appVersion: String
        get() {
            try {
                return ctx.packageManager.getPackageInfo(ctx.packageName, 0).versionName
            } catch (localNameNotFoundException: PackageManager.NameNotFoundException) {
            }
            return ""
        }
    val deviceId: String?
        get() {
            LogUtil.d("getDeviceId() : " + HNApplication.Companion.mDeviceId)
            return HNApplication.Companion.mDeviceId
        }
    var memberKey: String?
        get() = HNSharedPreference.getSharedPreference(ctx, "memberKey")
        set(paramString) {
            HNSharedPreference.putSharedPreference(ctx, "memberKey", paramString)
        }
    var appId: String?
        get() = HNSharedPreference.getSharedPreference(ctx, "appId")
        set(paramString) {
            HNSharedPreference.putSharedPreference(ctx, "appId", paramString)
        }
    var userId: String?
        get() = HNSharedPreference.getSharedPreference(ctx, "userId")
        set(paramString) {
            HNSharedPreference.putSharedPreference(ctx, "userId", paramString)
        }
    var channelId: String?
        get() = HNSharedPreference.getSharedPreference(ctx, "channelId")
        set(paramString) {
            HNSharedPreference.putSharedPreference(ctx, "channelId", paramString)
        }
    var requestId: String?
        get() = HNSharedPreference.getSharedPreference(ctx, "requestId")
        set(paramString) {
            HNSharedPreference.putSharedPreference(ctx, "requestId", paramString)
        }
}